examples/api/lib/material/input_decorator/input_decoration.widget_state.0.dart

```diff
@@ -32,13 +32,10 @@ class MaterialStateExample extends StatelessWidget {
       initialValue: 'abc',
       decoration: InputDecoration(
         prefixIcon: const Icon(Icons.person),
-        prefixIconColor: MaterialStateColor.resolveWith((Set<MaterialState> states) {
-          if (states.contains(MaterialState.focused)) {
+        prefixIconColor: WidgetStateColor.resolveWith((Set<WidgetState> states) {
+          if (states.contains(WidgetState.focused)) {
             return Colors.green;
           }
-          if (states.contains(MaterialState.error)) {
-            return Colors.red;
-          }
```

Comment:
Any reason for removing this case?
It is still mentioned line in packages/flutter/lib/src/material/input_decorator.dart line 2521.

Reply:
iirc there was no way to cause an error in the example, so I essentially just removed dead code. I now updated the doc comment in input_decorator to avoid confusion.

I don't think explicitly showing how to use the error state is of much use here as I already updated the `input_decoration.widget_state.1.dart` example to contain a validator.